### PR TITLE
Allow passing a reference composite to a context menu

### PR DESCRIPTION
**Pull request by apodhrad**

### CODE REVIEW

File: plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Control.java [hunk 1]
```diff
@@ -10,13 +10,15 @@
...
+import org.eclipse.reddeer.core.reference.ReferencedComposite;
+
 /**
  * API for control manipulation
  * @author rawagner
  *
  * @param <T> extends swt Control
  */
-public interface Control<T extends org.eclipse.swt.widgets.Control> extends Widget<T> {
+public interface Control<T extends org.eclipse.swt.widgets.Control> extends Widget<T>, ReferencedComposite {
```
Comment (rawagner) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Control.java]: Not every Control is not a ReferencedComposite. You can just add constructor to Menu which takes ReferencedComposite.

Comment (apodhrad) [reply plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Control.java]: No, I cannot. Assume the following example
```
new ContextMenu(new DefaultShelll("..."));
```
where the shell is Control and also ReferenceComposite

Comment (apodhrad) [reply plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Control.java]: @rawagner why Control cannot be a reference composite?

Comment (apodhrad) [reply plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Control.java]: Me and @rawagner agreed on adding a method
```
getContextMenu();
```
do editors and controls.

**Comment by apodhrad** [comment]
Rejected

**Pull request closed without merging**